pkg/notify/notify.go

```diff
@@ -3,166 +3,218 @@ package notify
 		if issue.Fields.Status.StatusCategory.Key != "done" {
-			// Issue is in a "to do" or "in progress" state, all done here.
-			level.Debug(logger).Log("msg", "issue is unresolved, nothing to do", "key", issue.Key, "label", issueLabel)
+			level.Debug(r.logger).Log("msg", "issue is unresolved, all is done", "key", issue.Key, "label", issueGroupLabel)
 			return false, nil
 		}
+
 		if r.conf.WontFixResolution != "" && issue.Fields.Resolution != nil &&
 			issue.Fields.Resolution.Name == r.conf.WontFixResolution {
-			// Issue is resolved as "Won't Fix" or equivalent, log a message just in case.
-			level.Info(logger).Log("msg", "issue was resolved as won't fix, not reopening", "key", issue.Key, "label", issueLabel, "resolution", issue.Fields.Resolution.Name)
+			level.Info(r.logger).Log("msg", "issue was resolved as won't fix, not reopening", "key", issue.Key, "label", issueGroupLabel, "resolution", issue.Fields.Resolution.Name)
 			return false, nil
 		}
 
-		resolutionTime := time.Time(issue.Fields.Resolutiondate)
-		if resolutionTime.Add(time.Duration(*r.conf.ReopenDuration)).After(time.Now()) {
-			level.Info(logger).Log("msg", "issue was recently resolved, reopening", "key", issue.Key, "label", issueLabel, "resolution_time", resolutionTime.Format(time.RFC3339), "reopen_duration", *r.conf.ReopenDuration)
-			return r.reopen(issue.Key, logger)
-		}
+		level.Info(r.logger).Log("msg", "issue was recently resolved, reopening", "key", issue.Key, "label", issueGroupLabel)
+		return r.reopen(issue.Key)
+	}
+
+	if len(data.Alerts.Firing()) == 0 {
+		level.Debug(r.logger).Log("msg", "no firing alert; nothing to do.", "label", issueGroupLabel)
+		return false, nil
+	}
+
+	level.Info(r.logger).Log("msg", "no recent matching issue found, creating new issue", "label", issueGroupLabel)
+
+	issueType, err := r.tmpl.Execute(r.conf.IssueType, data)
+	if err != nil {
+		return false, errors.Wrap(err, "render issue type")
+	}
+
+	issueDesc, err := r.tmpl.Execute(r.conf.Description, data)
+	if err != nil {
+		return false, errors.Wrap(err, "render issue description")
 	}
 
-	level.Info(logger).Log("msg", "no recent matching issue found, creating new issue", "label", issueLabel)
 	issue = &jira.Issue{
 		Fields: &jira.IssueFields{
 			Project:     jira.Project{Key: project},
-			Type:        jira.IssueType{Name: r.tmpl.Execute(r.conf.IssueType, data, logger)},
-			Description: r.tmpl.Execute(r.conf.Description, data, logger),
-			Summary:     r.tmpl.Execute(r.conf.Summary, data, logger),
-			Labels: []string{
-				issueLabel,
-			},
-			Unknowns: tcontainer.NewMarshalMap(),
+			Type:        jira.IssueType{Name: issueType},
+			Description: issueDesc,
+			Summary:     issueSummary,
+			Labels:      []string{issueGroupLabel},
+			Unknowns:    tcontainer.NewMarshalMap(),
 		},
 	}
 	if r.conf.Priority != "" {
-		issue.Fields.Priority = &jira.Priority{Name: r.tmpl.Execute(r.conf.Priority, data, logger)}
+		issuePrio, err := r.tmpl.Execute(r.conf.Priority, data)
```

Comment:
This just occurred to me now: what about updating the other fields? (In addition to the summary.) E.g. what if someone wants priority to be determined by the number of alerts firing? (E.g. 1-2 alerts: low priority; 3-5 alerts: medium priority; 6+ alerts: high priority.) Or any of the custom fields.

Not an immediate issue (as these fields were essentially immutable until now, even before the reopen feature) but may be worth considering.

Reply:
Fully agree. Created https://github.com/free/jiralert/issues/37